tokio/src/net/addr.rs

```diff
@@ -40,6 +40,61 @@ where
     }
 }
 
+cfg_dns! {
+    /// LookupHost contains resolved SocketAddrs.
+    ///
+    /// This type is created by the function [`lookup_host`]. See its documentation for more.
+    #[derive(Debug)]
+    pub struct LookupHost<T>
+    where
+        T: sealed::ToSocketAddrsPriv,
+        T: ToSocketAddrs<Future = sealed::MaybeReady>,
+        T: ToSocketAddrs<Iter = sealed::OneOrMore>,
+    {
+        fut: <T as sealed::ToSocketAddrsPriv>::Future,
+        addrs: Option<sealed::OneOrMore>
+    }
+
+    impl<T> LookupHost<T>
+    where
+        T: sealed::ToSocketAddrsPriv + Unpin,
+        T: ToSocketAddrs<Future = sealed::MaybeReady>,
+        T: ToSocketAddrs<Iter = sealed::OneOrMore>,
+    {
+        /// Fetches the next resolved address.
+        pub async fn next_addr(&mut self) -> Option<io::Result<SocketAddr>> {
```

Comment:
For an inherent fn, I believe `io::Result<Option<SocketAddr>>` is more natural. Thoughts?

Reply:
I agree! I was just following what you said in #1684 :).